lib/linguist/shebang.rb

```diff
@@ -10,10 +10,11 @@ class Shebang
     #
     # Returns an Array with one Language if the blob has a shebang with a valid
     # interpreter, or empty if there is no shebang.
-    def self.call(blob, _ = nil)
+    def self.call(blob, candidates)
       return [] if blob.symlink?
 
-      Language.find_by_interpreter interpreter(blob.data)
+      languages = Language.find_by_interpreter interpreter(blob.data)
+      if candidates.any? then candidates & languages else languages end
```

Comment:
By the way @lildude, do you know if there's a more Rubyist way to write this?

Reply:
Yup: 

```Ruby
candidates.any? ? candidates & languages : languages
```

Reply:
I think it looks fine with syntax highlighting, but if you'd prefer stick with the simpler syntax, tell me and I'll remove the last commit.